src/scenes/Activities/Favorite/store.js

```diff
@@ -0,0 +1,92 @@
+import { observable, action, runInAction, reaction } from 'mobx';
+import _ from 'lodash';
+import { Routes, SortBy, OracleStatus } from 'constants';
+import { queryAllTopics, queryAllOracles } from '../../../network/graphql/queries';
+import Topic from '../../../stores/models/Topic';
+import Oracle from '../../../stores/models/Oracle';
+
+const INIT_VALUES_FAVPAGE = {
+  loaded: false, // loading state?
+  loadingMore: true, // for loading icon?
+  displayList: [], // data list
+};
+
+export default class {
+  @observable favList = JSON.parse(localStorage.getItem('bodhi_dapp_favList')) || []; // Data example: '21e389b909c7ab977088c8d43802d459b0eb521a'
+
+  @observable loaded = INIT_VALUES_FAVPAGE.loaded
+  @observable displayList = INIT_VALUES_FAVPAGE.displayList
+  @observable loadingMore = INIT_VALUES_FAVPAGE.loadingMore
+
+  constructor(app) {
+    this.app = app;
+    reaction(
+      () => this.app.wallet.addresses + this.app.global.syncBlockNum,
+      () => {
+        if (this.app.ui.location === Routes.FAVORITE) {
+          this.init();
+        }
+      }
+    );
+    reaction(
+      () => this.displayList,
+      () => {
+      }
+    );
+    reaction(
+      () => this.favList,
+      async () => {
+        this.displayList = await this.fetchFav();
+      }
+    );
+  }
+
+  updateLocalStorageFavList() {
+    localStorage.setItem('bodhi_dapp_favList', JSON.stringify(this.favList));
+  }
+
+  @action
+  setFavorite = (topicAddress) => {
+    if (this.isInFavorite(topicAddress)) this.favList = this.favList.filter(x => x !== topicAddress);
+    else {
+      this.favList.push(topicAddress);
+      this.favList.replace(this.favList);
+    }
+    this.updateLocalStorageFavList();
+  }
+
+  @action
+  isInFavorite = (topicAddress) => this.favList.some(x => x === topicAddress)
+
+  @action
+  init = async () => {
+    Object.assign(this, INIT_VALUES_FAVPAGE); // reset to initial state
+    this.app.ui.location = Routes.FAVORITE; // change ui location, for tabs to render correctly
+    this.displayList = await this.fetchFav();
+  }
+
+  fetchFav = async () => {
+    if (this.favList.length === 0) return [];
+    // Get all event in WITHDRAW phase as Topic at favorite topic address list "favList"
+    const topicOrderBy = { field: 'endTime', direction: SortBy.ASCENDING };
+    const topicFilters = this.favList.map(topicAddress => ({ address: topicAddress, status: OracleStatus.WITHDRAW }));
+    const topics = await queryAllTopics(topicFilters, topicOrderBy, Infinity, Infinity);
```

Comment:
remove `skip` (the last param) since we dont want to skip any. and change `limit` to a big number like 5000. the default limit should be 50 so we need to increase it.

`Infinity` is not actually being used, `queryAllTopics` checks if those values are finite.